src/DynamicRupture/Output/ReceiverBasedOutput.cpp

```diff
@@ -0,0 +1,423 @@
+#include "ReceiverBasedOutput.hpp"
+#include "DynamicRupture/Output/OutputAux.hpp"
+#include "Initializer/tree/Layer.hpp"
+#include "Numerical_aux/BasisFunction.h"
+#include "generated_code/tensor.h"
+#include "generated_code/kernel.h"
+#include <unordered_map>
+
+namespace seissol::dr::output {
+void ReceiverBasedOutput::setLtsData(seissol::initializers::LTSTree* userWpTree,
+                                     seissol::initializers::LTS* userWpDescr,
+                                     seissol::initializers::Lut* userWpLut,
+                                     seissol::initializers::LTSTree* userDrTree,
+                                     seissol::initializers::DynamicRupture* userDrDescr) {
+  wpTree = userWpTree;
+  wpDescr = userWpDescr;
+  wpLut = userWpLut;
+  drTree = userDrTree;
+  drDescr = userDrDescr;
+}
+
+void ReceiverBasedOutput::getDofs(real dofs[tensor::Q::size()], int meshId) {
+  // get DOFs from 0th derivatives
+  assert((wpLut->lookup(wpDescr->cellInformation, meshId).ltsSetup >> 9) % 2 == 1);
+
+  real* derivatives = wpLut->lookup(wpDescr->derivatives, meshId);
+  std::copy(&derivatives[0], &derivatives[tensor::dQ::Size[0]], &dofs[0]);
+}
+
+void ReceiverBasedOutput::getNeighbourDofs(real dofs[tensor::Q::size()], int meshId, int side) {
+  real* derivatives = wpLut->lookup(wpDescr->faceNeighbors, meshId)[side];
```

Comment:
why is it called faceNeighbors and not neighborsDerivatives?

Reply:
It is the way how it was written in `Initializer/LTS.h`

https://github.com/SeisSol/SeisSol/blob/5d85df973f0b0af4052ccdd064c4424899521e16/src/Initializer/LTS.h#L87-L104

Reply:
ok, don't you think it would make more sense if called neighborsDerivatives?
we can fix that later one.

Reply:
We can but I would do it after the merge with the master branch. The most important thing now is to provide the same functionality as we have in the master branch and then merge.